helm-charts/infisical/templates/frontend-deployment.yaml

```diff
@@ -9,65 +10,69 @@ metadata:
+  replicas: {{ $frontend.replicaCount }}
   selector:
     matchLabels:
       {{- include "infisical.frontend.matchLabels" . | nindent 6 }}
   template:
     metadata: 
       labels:
         {{- include "infisical.frontend.matchLabels" . | nindent 8 }}
-      {{- with .Values.frontend.podAnnotations }}
+      {{- with $frontend.podAnnotations }}
       annotations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
     spec:
       containers:
-      - name: {{ template "infisical.name" . }}-{{ .Values.frontend.name }}
-        image: "{{ .Values.frontend.image.repository }}:{{ .Values.frontend.image.tag | default .Chart.AppVersion }}"
-        imagePullPolicy: {{ .Values.frontend.image.pullPolicy }}
+      - name: {{ template "infisical.name" . }}-{{ $frontend.name }}
+        image: "{{ $frontend.image.repository }}:{{ $frontend.image.tag | default "latest" }}"
+        imagePullPolicy: {{ $frontend.image.pullPolicy }}
         readinessProbe:
           httpGet:
             path: /
             port: 3000
           initialDelaySeconds: 10
           periodSeconds: 10
-        {{- if .Values.frontend.kubeSecretRef }}
         envFrom:
         - secretRef:
-            name: {{ .Values.frontend.kubeSecretRef }}
-        {{- end }}
-        {{- if .Values.frontendEnvironmentVariables }}
-        env:
-        {{- range $key, $value := .Values.frontendEnvironmentVariables }}
-        {{- if $value | quote | eq "MUST_REPLACE" }}
-          {{ fail "Environment variables are not set. Please set all environment variables to continue." }}
-        {{ end }}
-        - name: {{ $key }}
-          value: {{ quote $value }}
-        {{- end }}
-        {{- end }}
+            name: {{ $frontend.kubeSecretRef | default "infisical-frontend"  }}
         ports: 
         - containerPort: 3000
+
 ---
+
 apiVersion: v1
 kind: Service 
 metadata:
   name: {{ include "infisical.frontend.fullname" . }}
   labels:
     {{- include "infisical.frontend.labels" . | nindent 4 }}
-  {{- with .Values.frontend.service.annotations }}
+  {{- with $frontend.service.annotations }}
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
-  type: {{ .Values.frontend.service.type }}
+  type: {{ $frontend.service.type }}
   selector:
     {{- include "infisical.frontend.matchLabels" . | nindent 8 }}
   ports:
     - protocol: TCP
       port: 3000 # service
       targetPort: 3000 # container port 
-      {{- if eq .Values.frontend.service.type "NodePort" }}
-      nodePort: {{ .Values.frontend.service.nodePort }}
-      {{- end }}
\ No newline at end of file
+      {{- if eq $frontend.service.type "NodePort" }}
+      nodePort: {{ $frontend.service.nodePort }}
+      {{- end }}
+
+---
+
+{{ if not $frontend.kubeSecretRef }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $frontend.kubeSecretRef | default "infisical-frontend" }}
+type: Opaque
+data:
+  {{- range $key, $value := $frontend.secrets }}
+  {{ $key }}: {{ $value | b64enc }}
```

Comment:
I think we should wrap the value in quotes before `b64enc` otherwise we get 

```
rror: UPGRADE FAILED: template: infisical/templates/frontend-deployment.yaml:76:26: executing "infisical/templates/frontend-deployment.yaml" at <b64enc>: wrong type for value; expected string; got float64
```